### PR TITLE
Maiorano pr 597 review

### PR DESCRIPTION
**Pull request by pioneerskies**
Hello @maiorano84 :)

It was too hard to discuss on the improvement I'd liked to introduce in specs, so I opted to open up a PR here. Take a look at my loing comment on 5af6791 , but I'll leave some comment on code too.

### CODE REVIEW

File: spec/sql_adapter/wpcli_spec.rb [hunk 1]
```diff
@@ -12,44 +14,39 @@
...
-  end
-
-  context "#command_with_params" do
-    before do
-      allow(adapter).to receive(:wp_in_path?).and_return(true)
-      allow(adapter).to receive(:`).and_return("{\"path\":{\"current\":\"\/path\/to\/pudding\"}}")
-    end
+  context "#command" do
+    context "having wp-cli.yml in local_path" do
+      let(:local_path) { fixture_folder_root_relative_path }
```
Comment (pioneerskies) [inline spec/sql_adapter/wpcli_spec.rb:19]: This way we'll really let `load_from_yml` method doing its job

File: spec/sql_adapter/wpcli_spec.rb [hunk 2]
```diff
@@ -12,44 +14,39 @@
...
-  context "#command_with_no_configured_path" do
-    before do
-      allow(adapter).to receive(:wp_in_path?).and_return(true)
-      allow(File).to receive(:exist?).and_return(false)
-      allow(adapter).to receive(:`).and_return("{}")
+    context "without wp-cli.yml in local_path" do
+      before do
+        allow(adapter)
+          .to receive(:`)
+          .with('wp cli param-dump --with-values')
```
Comment (pioneerskies) [inline spec/sql_adapter/wpcli_spec.rb:32]: I've narrowed the scope of the stub to the specific command, just to

* be sure we're not stubbing every requests (maybe during future development we'll do anote call to the "bscktick" method
* it's more evident in the test what we're stubbing

**Comment by pioneerskies** [comment]
He he, I fucked up a test. I'll check it ASAP ;)

**Comment by maiorano84** [comment]
@pioneerskies 

My approach with unit testing tends to involve a LOT of mocking to ensure that all unit tests are running in isolation, and all public methods aren't leaking into other "live" objects. It's a concept that carried over from my usual unit testing workflows in PHP.

My approach is almost certainly overblown and unnecessary for most projects, given the nature of my PR and the actual change. I'm 100% on board with you simplifying this. Once the checks are passing, I'll merge your PR in.

**Comment by pioneerskies** [comment]
> @pioneerskies
> 
> My approach with unit testing tends to involve a LOT of mocking to ensure that all unit tests are running in isolation, and all public methods aren't leaking into other "live" objects. It's a concept that carried over from my usual unit testing workflows in PHP.

Just for the sake of discussion: isolating is ok, but isolation should be mostly achieved through dependency injection; so if your method is already isolated, you are free to stub anything passed to it through arguments. But if you need to stub what's inside a method I think that's a smell. For sure Wordmove suffers from a poor design in many methods; but that's a fact. And to deal with it, e.g. having a method which is accessing files inside of it, I prefere, if it's not too much effort, to fixturize those files and actually have the method accessing them.

My rule of thumb would be: stub what's outside the method, not what's inside, whenever possible :)

> Another note is that I was probably over-mocking due to the fact that I'm on Windows and the underlying `system` calls were failing for me. I didn't feel like trying to figure out setting up an underlying Ruby Container, so I figured that the mocks would be more predictable.

This is intresting; I'd like to know if my refactor will run on your system too.

> My approach is almost certainly overblown and unnecessary for most projects, given the nature of my PR and the actual change. I'm 100% on board with you simplifying this. Once the checks are passing, I'll merge your PR in.

Now they seems to be all green.

Cheers!

**Comment by maiorano84** [comment]
All set, man! Thanks for looking at this.

**Pull request merged**